packages/chrono-core/src/processors/simple-processor.ts

```diff
@@ -192,15 +178,42 @@ export class SimpleProcessor<
 
   private async handleTaskError(task: Task<TaskKind, TaskMapping[TaskKind]>, error: Error): Promise<void> {
     const failedAt = new Date();
+
     if (task.retryCount >= this.config.taskHandlerMaxRetries) {
-      // Mark the task as failed
-      await this.datastore.fail(task.id);
+      // --- Existing behavior: mark the task as failed ---
+      // Use DLQ if available, otherwise fallback to fail
+      if (this.datastore.addToDlq) {
+        await this.datastore.addToDlq(task, error);
+      } else {
+        await this.datastore.fail(task.id);
+      }
+
       this.emit(ProcessorEvents.TASK_FAILED, {
         task,
         error,
         failedAt,
       });
 
+      // --- Optional redrive logic ---
+      if (this.datastore.redriveFromDlq) {
+        const redriveDelayMs = 60_000; // 1 minute
+        await setTimeout(redriveDelayMs); // Delay before redrive
+        try {
+          await this.datastore.redriveFromDlq<TaskKind>();
+          this.emit(ProcessorEvents.TASK_RETRY_SCHEDULED, {
+            task,
+            error: null,
+            errorAt: new Date(),
+            retryScheduledAt: new Date(Date.now() + redriveDelayMs),
+          });
+        } catch (redriveError) {
+          this.emit(ProcessorEvents.UNKNOWN_PROCESSING_ERROR, {
+            error: redriveError,
+            timestamp: new Date(),
+          });
+        }
+      }
```

Comment:
slightly confused by this 
1. you are blocking the processing-loop for 60 seconds. meaning no new tasks can be claimed while this flow is being hit
2. redriveFromDlq grabs all "tasks" in the DLQ yet you only emit a single TASK_RETRY_SCHEDULED event despite the possibility of many more being "re-scheduled" 
3. this logic adds a task document to the DLQ collection. Queries it. Inserts identical document into main collection. but does not update the "old" document in any way. And because you have now by-passed the `await this.datastore.fail` call the original document will be re-queried after the timeout interval.